spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/DefaultChannelFactory.java

```diff
@@ -30,21 +29,33 @@
  */
 public class DefaultChannelFactory implements ChannelFactory {
 
-	@Autowired
-	MessageConverterConfigurer messageConverterConfigurer;
+	private final MessageConverterConfigurer messageConverterConfigurer;
+
+	public DefaultChannelFactory(MessageConverterConfigurer messageConverterConfigurer) {
+		this.messageConverterConfigurer = messageConverterConfigurer;
+	}
 
 	@Override
-	public MessageChannel createChannel(String name, Class<?> inputChannelType) throws Exception {
-		MessageChannel messageChannel = createMessageChannel(inputChannelType);
-		messageConverterConfigurer.configureMessageConverters(messageChannel, name);
-		return messageChannel;
+	public PollableChannel createPollableBindableChannel(String name) throws Exception {
+		PollableChannel pollableChannel = new QueueChannel();
+		messageConverterConfigurer.configureMessageConverters(pollableChannel, name);
+		return pollableChannel;
 	}
 
-	private MessageChannel createMessageChannel(Class<?> messageChannelType) {
-		return isPollable(messageChannelType) ? new QueueChannel() : new DirectChannel();
+	@Override
+	public SubscribableChannel createSubscribableBindableChannel(String name) throws Exception {
+		SubscribableChannel subscribableChannel = new DirectChannel();
+		messageConverterConfigurer.configureMessageConverters(subscribableChannel, name);
+		return subscribableChannel;
 	}
 
-	private boolean isPollable(Class<?> channelType) {
-		return PollableChannel.class.equals(channelType);
+	@Override
+	public SubscribableChannel createSubscribableSharedChannel() throws Exception {
```

Comment:
Just to clarify you are suggesting that we only keep the methods that *do* apply configuration but removing "bindable" from those method names, and then completely removing the 2 methods with "shared" in the name. The responsibility for shared channel creation (which is just a `new` call) would go back to the `BindableProxyFactory` (to be renamed), and the bridging logic would remain in that class also. Is that correct?